src/overlays/actors/ovl_En_Wiz/z_en_wiz.c

```diff
@@ -30,199 +87,1467 @@ const ActorInit En_Wiz_InitVars = {
+            this->actor.world.rot.y = angle;
+            for (i = 0; i < this->platformCount; i++) {
+                this->ghostRot[i].y = Math_Vec3f_Yaw(&this->ghostPos[i], &player->actor.world.pos);
+            }
+
+            EnWiz_ChangeAnim(this, EN_WIZ_ANIM_IDLE, true);
+            this->shouldStartTimer = false;
+            this->targetPlatformLightAlpha = 255;
+            Math_Vec3f_Copy(&this->platformLightPos, &this->actor.world.pos);
+
+            if (this->fightState == EN_WIZ_FIGHT_STATE_FIRST_PHASE) {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_WIZ_UNARI);
+            } else {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_WIZ_VOICE - SFX_FLAG);
+            }
+        }
+
+        this->actionFunc = EnWiz_Appear;
+    }
+}
+
+/**
+ * Makes the Wizrobe grow in scale and become more opaque.
+ * During the first phase, it also increases the alpha for the platform's light.
+ * If the player gets too close during the first phase, this will set up the Wizrobe to disappear.
+ */
+void EnWiz_Appear(EnWiz* this, PlayState* play) {
+    Vec3f staffTargetFlameScale = { 0.006f, 0.006f, 0.006f };
+    Player* player = GET_PLAYER(play);
+
+    EnWiz_HandleIntroCutscene(this, play);
+
+    if (this->introCutsceneState >= EN_WIZ_INTRO_CS_APPEAR) {
+        SkelAnime_Update(&this->skelAnime);
+
+        if ((this->fightState == EN_WIZ_FIGHT_STATE_FIRST_PHASE) &&
+            (this->introCutsceneState >= EN_WIZ_INTRO_CS_DISAPPEAR) &&
+            ((this->actor.xzDistToPlayer < 200.0f) ||
+             ((player->unk_D57 != 0) &&
+              ((ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, this->actor.shape.rot.y)) < 0x7D0)) &&
+              (ABS_ALT(BINANG_SUB(this->actor.yawTowardsPlayer, BINANG_ADD(player->actor.shape.rot.y, 0x8000))) <
+               0x7D0)))) {
+            EnWiz_SetupDisappear(this);
+        } else {
+            Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 0xA, 0xBB8, 0);
+
+            if (this->fightState == EN_WIZ_FIGHT_STATE_FIRST_PHASE) {
+                Math_SmoothStepToS(&this->platformLightAlpha, this->targetPlatformLightAlpha, 10, 10, 10);
+                if (!this->shouldStartTimer) {
+                    this->timer = 20;
+                    this->shouldStartTimer = true;
+                }
+            } else {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_WIZ_VOICE - SFX_FLAG);
+            }
+
+            if (this->timer == 0) {
+                Math_ApproachF(&this->scale, 0.015f, 0.05f, 0.01f);
+                Math_SmoothStepToS(&this->alpha, 255, 1, 5, 0);
+            }
+
+            if (this->scale < 0.0138f) {
+            } else {
+                this->action = EN_WIZ_ACTION_RUN_IN_CIRCLES;
+                this->actor.flags &= ~ACTOR_FLAG_8000000;
+                this->ghostColliders.elements[0].info.bumper.dmgFlags =
+                    0x1000000 | 0x10000 | 0x2000 | 0x1000 | 0x800 | 0x200 | 0x20 | 0x2;
```

Comment:
It's easier to process this later as one lump, I don't see the merit in writing it this way, especially when there are so many.